gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceExtractor.java

```diff
@@ -871,8 +871,10 @@ public SalesforceBulkJobId getQueryResultIdsPkChunking(String entity, List<Predi
 
       // Wait for pk chunking batches
       BatchInfoList batchInfoList = this.bulkConnection.getBatchInfoList(this.bulkJob.getId());
-
-      if (bulkBatchInfo.getState() == BatchStateEnum.Failed) {
+      BatchStateEnum state = bulkBatchInfo.getState();
+      // not sure if the state can be "NotProcessed" in non-pk-chunking bulk request
+      // SFDC doc says - This state is assigned when a job is aborted while the batch is queued
+      if (state == BatchStateEnum.Failed || state == BatchStateEnum.InProgress) {
```

Comment:
Do you mean `BatchStateEnum.NotProcessed` instead of `InProgress` here? Should never reach here with `InProgress` state because of the loop above.